### PR TITLE
HOTFIX - Eislam/sy 1064 ni scanner touchups

### PR DESCRIPTION
**Pull request by Lham42**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-1064]()

## Description

This PR makes a couple improvements to error handling in the ni scanner. Mostly makes sure that when properties aren't found for certain devices, the scanner still remains functional and when there is an issue, it's error state is stored for future use of the scanner.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added sufficient regression tests to cover the changes to CI.
- [x] I have added relevant tests to cover the changes or exposing bugs.
- [x] I have verified code coverage targets are met.

## Additional Notes
- [ ] These changes deal with concurrency.
- [ ] These changes affect UI.

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

## Reviewer Checklist
- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in readiness checklists.
- [ ] UI changes have been tested.
- [ ] style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.

### CODE REVIEW

File: driver/main.cpp [hunk 1]
```diff
@@ -45,6 +45,8 @@ std::pair<synnax::Rack, freighter::Error> retrieveDriverRack(
 std::atomic<bool> stopped = false;
 
 int main(int argc, char *argv[]) {
+    //sleep for 5 seconds
+    std::this_thread::sleep_for(std::chrono::seconds(5));
```
Comment (emilbon99) [inline driver/main.cpp:49]: Remove this sleep.

File: driver/ni/scanner.cpp [hunk 2]
```diff
@@ -33,12 +33,7 @@ ni::Scanner::Scanner(
         &this->session // session handle
     );
 
-    if (status != NISysCfg_OK) {
-        this->ok_state = false;
-        LOG(ERROR) << "[ni.scanner] failed to initialize scanner for task " << this->
-                task.name;
-        return;
-    }
+    if (status != NISysCfg_OK) log_err("failed to initialize scanner");
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:36]: Is it ok that we got rid of the early return here?

File: driver/ni/scanner.cpp [hunk 3]
```diff
@@ -84,12 +79,8 @@ void ni::Scanner::scan() {
         this->filter, NULL,
         &this->resources_handle
     );
-    if (err != NISysCfg_OK) {
-        this->ok_state = false;
-        LOG(ERROR) << "[ni.scanner] failed to find hardware for task " << this->task.
-                name;
-        return;
-    }
+    if (err != NISysCfg_OK) return log_err("failed to find hardware");
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:82]: Again, is it ok that we got rid of the early return?

File: driver/ni/scanner.cpp [hunk 4]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
 
-
 json ni::Scanner::get_device_properties(NISysCfgResourceHandle resource) {
     json device;
 
     char propertyValue[1024] = "";
+    int status;
 
-    ni::NiSysCfgInterface::GetResourceProperty(
+    // serial number
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:104]: Useless commant

File: driver/ni/scanner.cpp [hunk 5]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
+    status = ni::NiSysCfgInterface::GetResourceProperty(
         resource,
         NISysCfgResourcePropertySerialNumber,
         propertyValue
     );
+    if (status != NISysCfg_OK) log_err("failed to get serial number");
     device["serial_number"] = propertyValue;
 
-    ni::NiSysCfgInterface::GetResourceProperty(
+    // product name
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:113]: Useless comment

File: driver/ni/scanner.cpp [hunk 6]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
         propertyValue
     );
+    if (status != NISysCfg_OK) log_err("failed to get product name");
     std::string model = propertyValue;
-    model = model.substr(3, model.size());
+    if (model.size() > 3) model = model.substr(3); 
     device["model"] = model;
 
-    ni::NiSysCfgInterface::GetResourceIndexedProperty(
+    // location
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:124]: Useless comment

File: driver/ni/scanner.cpp [hunk 7]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
         NISysCfgIndexedPropertyExpertUserAlias,
         0,
         propertyValue
     );
+    if (status != NISysCfg_OK) log_err("failed to get location");
     device["location"] = propertyValue;
 
-    ni::NiSysCfgInterface::GetResourceIndexedProperty(
+
+    // resource name
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:135]: Useless comment

File: driver/ni/scanner.cpp [hunk 8]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
     std::string rsrc_name = propertyValue;
-    rsrc_name = rsrc_name.substr(1, rsrc_name.size() - 2);
+    if (rsrc_name.size() > 2) rsrc_name = rsrc_name.substr(1, rsrc_name.size() - 2);
+    else  log_err("resource name too short to extract name");
     device["resource_name"] = rsrc_name;
-
-    double temp;
-    ni::NiSysCfgInterface::GetResourceProperty(
+    
+    // temperature
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:148]: Useless comment

File: driver/ni/scanner.cpp [hunk 9]
```diff
@@ -104,74 +95,84 @@ void ni::Scanner::scan() {
...
+    double temp = 0;
+    status = ni::NiSysCfgInterface::GetResourceProperty(
         resource,
         NISysCfgResourcePropertyCurrentTemp,
         &temp
     );
     device["temperature"] = temp;
 
+    // whether it is a simulated device
```
Comment (emilbon99) [inline driver/ni/scanner.cpp:157]: Useless comment

**Review by emilbon99 — APPROVED** [review]
Remove the sleep from main and make sure the places where you got rid of an early return can't cause problems like a bad access.

File: driver/ni/scanner.cpp (hunk 3 above)
Comment (Lham42) [reply driver/ni/scanner.cpp:82]: this one still returns

**Comment by emilbon99** [comment]
@Lham42 Build fails

**Pull request merged**